src/test/java/NewBalance/Tests/NewBalancePriceFilterTest.java

```diff
@@ -0,0 +1,51 @@
+package NewBalance.Tests;
+
+import BasePageAndTestInit.TestInit;
+import NewBalance.Pages.NewBalanceHeaderViewHelper;
+import NewBalance.Pages.NewBalanceManShoesPageHelper;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+
+public class NewBalancePriceFilterTest extends TestInit {
+
+    public void waitUrlContains(String text) {
```

Comment:
Put it in BasePage please, and then use to find Xpath in method
    public WebElement get**NameOfYourElement**() {
        return getElement**WithUrlContainsWait**(_**STRING_LOCATOR**_);
    }

Reply:
I understand, but this test is not finished, cuse it has these problems